xsplice: Tag xsplice_work __read_mostly

This variable is check in the idle loop and return to guest
code paths, so put it in __read_mostly to avoid cacheline
bouncing.

diff --git a/xen/common/xsplice.c b/xen/common/xsplice.c
index 16c6609..0b08153 100644
--- a/xen/common/xsplice.c
+++ b/xen/common/xsplice.c
@@ -85,7 +85,7 @@ struct xsplice_work
 };
 
 /* There can be only one outstanding patching action. */
-static struct xsplice_work xsplice_work;
+static struct xsplice_work __read_mostly xsplice_work;
 
 static int schedule_work(struct payload *data, uint32_t cmd, uint32_t timeout);
 
